### ISSUE TITLE
Problem cloning nested submodules

### ISSUE THREAD

**Issue by antibaben** [rootiest/dotfiles#2]
- Descrption: While installing, I get the multiple errors describing no being able to find submodule URLs for the submodule tmux, like for tpm as is seen in the following error output.

```
Submodule path 'home/dot_config/tmux': checked out '70da64e637c97b78071f641ff2a27f1dcb15c387'
fatal: No url found for submodule path 'home/dot_config/tmux/.tmux/plugins/tpm' in .gitmodules
fatal: Failed to recurse into submodule path 'home/dot_config/tmux'
chezmoi: git: exit status 128
```

- Command run: `sh -c "$(curl -fsLS git.io/chezmoi)" -- init --apply rootiest`
- Expected result: succesful instalation of rootiest dotfiles.

**Comment by rootiest** [comment]
Hey sorry for the delay on getting to this.

I have an idea of the problem though, I think I simply need to tag the `tmux` directory as `external_` as noted [here in the chezmoi docs](https://www.chezmoi.io/user-guide/include-files-from-elsewhere/#use-git-submodules-in-your-source-directory)

> If you use git submodules, then you should set the external_ attribute on the subdirectory containing the submodule.

I just moved so some of my testing system is not set up but I will push an update that will hopefully correct this issue if you are able to/want to give it a shot!

Assuming it goes well, I will also probably do the same with the other submodule directories as it seems like I probably should be.

**Comment by rootiest** [comment]
Here is the commit:

https://github.com/rootiest/dotfiles/commit/cc729a5509915dcb3edf97ff5b40843c6cf95ef3 [fix: correct submodule path in configuration file](https://github.com/rootiest/dotfiles/commit/cc729a5509915dcb3edf97ff5b40843c6cf95ef3)

**Comment by rootiest** [comment]
Annnnndd forget all of that. 😅

I am like 99% sure I found the solution and it's on the tmux side.

What I did was delete the plugins dir, add the plugins dir to .gitignore, and then added a little snippet in the tmux config to identify when tmux plugin manager is not yet installed and install it automatically before using it.

I think that should correct the issue without chezmoi/my dotfiles config needing any changes.

The relevant commit is here: https://github.com/rootiest/rootiest-tmux/commit/e60c9df11ae9f8cf570033c861708fa7c73e55b5 with the important lines being:

```
# Bootstrap TMUX plugin manager
if "test ! -d ~/.config/tmux/plugins/tpm" \
   "run 'git clone https://github.com/tmux-plugins/tpm ~/.config/tmux/plugins/tpm && ~/.config/tmux/plugins/tpm/bin/install_plugins'"
```

**Comment by antibaben** [comment]
Hi, thanks for the help. I think this might work. Can't test it though, external_tmux points to previous commit https://github.com/rootiest/rootiest-tmux/commit/70da64e637c97b78071f641ff2a27f1dcb15c387, not the one you mentioned.

**Comment by antibaben** [comment]
Hi, sorry to bother you again. Any update on this? The reported error keeps happening and the fix mentioned can't be tested.

**Comment by rootiest** [comment]
No worries! Not a bother!

That was my bad, I have such a hard time with submodules haha

I think it should be pointing at the correct commit now 😅

**Comment by antibaben** [comment]
Thanks. It is pointing to the correct commit now. Sadly, the error continues. I don't understand much about submodules.
```
Submodule path 'home/dot_config/external_tmux': checked out 'e60c9df11ae9f8cf570033c861708fa7c73e55b5'
fatal: No url found for submodule path 'home/dot_config/external_tmux/.tmux/plugins/tpm' in .gitmodules
fatal: Failed to recurse into submodule path 'home/dot_config/external_tmux'
chezmoi: git: exit status 128
```
I really wanted to give this a try :/

**Comment by rootiest** [comment]
Gah submodules why you hate me so much!!! :sob: 

The worst part is my tmux config is really basic and only really needs the one tmux.conf file to work

I tried again to fix it, maybe it will work now?

I may have to rethink how I have this all set up if submodules keep being the bane of my existence haha

**Comment by rootiest** [comment]
I'm sorry this is so broken but I will keep working at it!

**Comment by antibaben** [comment]
Hi, I've managed to complete the installation, but nothing happened. Could I've missed some step? I know this doesn't have to do with the topic, but I don't want to open another issue for something that might be too small for a new one. After cloning, I got a set of questions that I answered ¿were any of the questions required to install the configurations? Because I don't know what is a GitHub token nor a GitHub Guard, so I did not pass any value when answering those questions.

**Comment by rootiest** [comment]
> Because I don't know what is a GitHub token nor a GitHub Guard, so I did not pass any value when answering those questions.

Those questions are used to fill in "private data"

For example, the .gitconfig file that sets up the `git` tool:
https://github.com/rootiest/dotfiles/blob/main/home/dot_gitconfig.tmpl
will use your github credentials that you are meant to provide in those questions to fill in the template there.

That way you can use the `git` program to push to your private repos/etc.

It's a bit confusing to read but this file is the one responsible for asking those questions at start:
https://github.com/rootiest/dotfiles/blob/main/home/.chezmoi.toml.tmpl

It collects that data and then uses it to complete the templates for any configuration files that would normally contain such private data (like a github token) that you (and I) would obviously not want to push to a public repo like this one.

Essentially it allows my dotfiles repo to contain files like `.gitconfig` that contain some configurations that make sense to push publicly like the githooks and stuff that are part of my dotfiles config, while leaving out the parts I definitely don't want to accidentally include (such as my personal github credentials)

Then the chezmoi template fills in those private details using the data provided in the setup questions.

[Github tokens are explained here](docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens)

**Comment by antibaben** [comment]
Right, I get what you explain now, but, does this have to do with how I don't see any changes after installation? Maybe I misunderstood what this tool does, because I was under the impression that I would get some visual changes in my desktop environment and some tools installed, like kitty, tmux and others, but I didn't get any. Could this have anything to do with how I've attempted to install this tool multiple times and if so, is there a way to clean the system to do a completely new installation? By the way, I'm running this in an arch Linux + KDE machine.

**Comment by rootiest** [comment]
Ah, I see.

Yeah it's possible the package installer part is broken, I never really perfected it.

You ***should*** now at least have my configs for those applications/tools but you may have to install the applications/tools themselves manually.

Sorry about that, this is definitely still a work in progress for me :disappointed:

**Comment by antibaben** [comment]
Ah, okay. No problem then. Thanks, by the way, for your work and answers.
